sdk/core/Azure.Core.Extensions/samples/CustomPolicy.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Pipeline;
+using Microsoft.AspNetCore.Hosting;
+
+namespace Azure.Core.Extensions.Samples
+{
+    internal class DIEnabledPolicy : SynchronousHttpPipelinePolicy
```

Comment:
You may want to consider spelling out Dependency Injection or otherwise renaming.  The triple-caps-in-a-row took me a minute and rubs against common convention.